Objects/longobject.c

```diff
@@ -378,32 +380,51 @@ PyLong_FromLong(long ival)
     return (PyObject *)v;
 }
 
+#define PYLONG_FROM_UINT(INT_TYPE, ival) \
+    do { \
+        if (IS_SMALL_UINT(ival)) { \
+            return get_small_int((ival)); \
+        } \
+        /* Count the number of Python digits. */ \
+        Py_ssize_t ndigits = 0; \
+        INT_TYPE t = (ival); \
+        while (t) { \
+            ++ndigits; \
+            t >>= PyLong_SHIFT; \
+        } \
+        PyLongObject *v = _PyLong_New(ndigits); \
+        if (v != NULL) { \
```

Comment:
Would you mind to write instead:

```
if (v == NULL) {
    return NULL;
}
```

I dislike long `if (... != NULL) { ... } return ...;` blocks which add one indentation level.